src/foam/dao/AuthenticatedDAO.java

```diff
@@ -99,27 +102,33 @@ public FObject find_(X x, Object id) {
   public Sink select_(X x, Sink sink, long skip, long limit, Comparator order, Predicate predicate) {
     // x.get("auth") != null added so that system user can act during booting when the auth service is not loaded yet.
     if ( authenticateRead_ && x.get("auth") != null ) {
```

Comment:
If you move the authenticateRead_ && x.get()... check into augmentPredicate, it will make the code simpler and fix listen_.

Reply:
We don't want the `authenticateRead_` check to apply to `removeAll_`. I can move just the `x.get("auth") != null` check though.

Reply:
If I just remove the `x.get("auth") != null` check altogether, the server still runs and the tests still pass. Maybe whatever failure this was put in to fix has been fixed elsewhere since this was put in. Will just remove it instead.

Reply:
Resolved by 02d6fa3.